cardano-node/src/Cardano/Node/Protocol/Cardano.hs

```diff
@@ -137,51 +134,81 @@ mkConsensusProtocolCardano NodeByronProtocolConfiguration {
-        (Consensus.TriggerHardForkAtVersion 3)
-
+        Consensus.ProtocolParamsByron {
+          byronGenesis = byronGenesis,
+          byronPbftSignatureThreshold =
+            PBftSignatureThreshold <$> npcByronPbftSignatureThresh,
+          byronProtocolVersion =
+            Byron.ProtocolVersion
+              npcByronSupportedProtocolVersionMajor
+              npcByronSupportedProtocolVersionMinor
+              npcByronSupportedProtocolVersionAlt,
+          byronSoftwareVersion =
+            Byron.SoftwareVersion
+              npcByronApplicationName
+              npcByronApplicationVersion,
+          byronLeaderCredentials =
+            byronLeaderCredentials
+        }
+        Consensus.ProtocolParamsShelley {
+          shelleyGenesis = shelleyGenesis,
+          shelleyInitialNonce =
+            Shelley.genesisHashToPraosNonce shelleyGenesisHash,
+          shelleyProtVer =
+            ProtVer
+              npcShelleySupportedProtocolVersionMajor
+              npcShelleySupportedProtocolVersionMinor,
+          shelleyLeaderCredentials =
+            shelleyLeaderCredentials
+        }
+        Consensus.ProtocolParamsAllegra {
+          allegraProtVer = error "TODO @node team: get from config file",
+          -- TODO @node team: get from config file
+          allegraLeaderCredentials = Nothing
+        }
+        Consensus.ProtocolParamsMary {
+          maryProtVer = error "TODO @node team: get from config file",
+          -- TODO @node team: get from config file
+          maryLeaderCredentials = Nothing
+        }
+        -- Byron to Shelley hard fork parameters
+        Consensus.ProtocolParamsTransition {
+          transitionLowerBound = npcShelleyHardForkNotBeforeEpoch,
+          transitionTrigger =
+            -- What will trigger the Byron->Shelley hard fork?
+            case npcTestShelleyHardForkAtEpoch of
+
+               -- This specifies the major protocol version number update that will
+               -- trigger us moving to the Shelley protocol.
+               --
+               -- Version 0 is Byron with Ouroboros classic
+               -- Version 1 is Byron with Ouroboros Permissive BFT
+               -- Version 2 is Shelley
+               --
+               -- But we also provide an override to allow for simpler test setups
+               -- such as triggering at the 0 -> 1 transition .
+               --
+               Nothing -> Consensus.TriggerHardForkAtVersion
+                            (maybe 2 fromIntegral npcTestShelleyHardForkAtVersion)
+
+               -- Alternatively, for testing we can transition at a specific epoch.
+               --
+               Just epochNo -> Consensus.TriggerHardForkAtEpoch epochNo
+        }
+        -- Shelley to Allegra hard fork parameters
+        Consensus.ProtocolParamsTransition {
+          -- TODO @node team: get from config file
+          transitionLowerBound = Nothing,
```

Comment:
This `transitionLowerBound` and the one below it are not the most urgent. Consensus might remove them at some point (https://github.com/input-output-hk/ouroboros-network/issues/2646).